### ISSUE TITLE
Object literal properties checked against a contextual type don't get `@deprecated` suggestions

### ISSUE THREAD

**Issue by u9g** [microsoft/typescript-go#4294]
### Version

7.0.0-dev.20260612.1 (also present in the JS checker, all versions since 4.0)

### Repro

```ts
interface RequestOptions {
    /** @deprecated use signal instead */
    timeout?: number;
}
declare function request(url: string, opts: RequestOptions): void;

request("/api", { timeout: 5000 });  // no deprecation suggestion
declare const opts: RequestOptions;
opts.timeout;                        // 'timeout' is deprecated. (6385) ✅
```

Verified against today's native preview via LSP `textDocument/diagnostic`: the property
read on the last line is reported (hint, code 6385), the object literal property on the
`request(...)` call is not. Editors therefore never show the deprecation strikethrough
in the most common place deprecated options are actually written — options-bag call
sites.

### Prior history in the JS repo

- microsoft/TypeScript#45657 — suggestion for exactly this (open, In Discussion)
- microsoft/TypeScript#39374 — umbrella bug; the JSX slice was fixed by #50084,
  the object literal slice never landed
- microsoft/TypeScript#46167 — a previous attempt, closed. The review blocker:
  `checkObjectLiteral` runs once per overload candidate, so reporting from inside it
  flags a deprecated property on *any* candidate's parameter type, even when a
  non-deprecated overload wins.

*Disclosure: the investigation behind this report was done with AI assistance (Claude
Code); I've reviewed and understand the findings and will be responding to feedback
myself.*
